contracts/forger_stake_v2/js/src/index.js

```diff
@@ -0,0 +1,322 @@
+  console.log("\nSending transaction...")
+  await web3.eth.sendSignedTransaction(signedTransaction.rawTransaction).then(console.log).catch(console.error);
+}
+
+const getCurrentConsensusEpoch = async () => {
+  console.log("\nRunning getCurrentConsensusEpoch script...");
+  validateEnvVars(
+    "getCurrentConsensusEpoch",
+    ["NETWORK"]
+  );
+  const {contract} = initializeWeb3AndContract();
+  await contract.methods.getCurrentConsensusEpoch().call().then(console.log);
+}
+
+const delegate = async () => {
+  console.log("\nRunning delegate script...")
+  validateEnvVars(
+    "delegate",
+    [
+      "NETWORK",
+      "MAX_PRIORITY_FEE_PER_GAS",
+      "MAX_FEE_PER_GAS",
+      "DELEGATE_FROM_ADDRESS",
+      "DELEGATE_PRIVATE_KEY",
+      "DELEGATE_BLOCK_SIGN_PUBLIC_KEY",
+      "DELEGATE_VRF_PUBLIC_KEY",
+      "DELEGATE_AMOUNT"
+    ]
+  );
+  const {contract, web3} = initializeWeb3AndContract();
+  const address = process.env.DELEGATE_FROM_ADDRESS;
+  const privateKey = process.env.DELEGATE_PRIVATE_KEY;
+  const value = web3.utils.toWei(process.env.DELEGATE_AMOUNT, "ether");
+  const {
+    blockSignPublicKey,
+    first32BytesForgerVrfPublicKey,
+    lastByteForgerVrfPublicKey
+  } = prepareForgerKeys(process.env.DELEGATE_BLOCK_SIGN_PUBLIC_KEY, process.env.DELEGATE_VRF_PUBLIC_KEY);
+  const data = contract.methods.delegate(blockSignPublicKey, first32BytesForgerVrfPublicKey, lastByteForgerVrfPublicKey).encodeABI();
+  await signAndSend(web3, contract, address, privateKey, data, value);
+}
+
+const withdraw = async () => {
+  console.log("\nRunning withdraw script...")
+  validateEnvVars(
+    "withdraw",
+    [
+      "NETWORK",
+      "MAX_PRIORITY_FEE_PER_GAS",
+      "MAX_FEE_PER_GAS",
+      "GAS_LIMIT",
+      "WITHDRAW_FROM_ADDRESS",
+      "WITHDRAW_PRIVATE_KEY",
+      "WITHDRAW_BLOCK_SIGN_PUBLIC_KEY",
+      "WITHDRAW_VRF_PUBLIC_KEY",
+      "WITHDRAW_AMOUNT"
+    ]
+  );
+  const {contract, web3} = initializeWeb3AndContract();
+  const address = process.env.WITHDRAW_FROM_ADDRESS;
+  const privateKey = process.env.WITHDRAW_PRIVATE_KEY;
+  const value = web3.utils.toWei(process.env.WITHDRAW_AMOUNT, "ether");
+  const gasLimit = process.env.GAS_LIMIT;
+  const {
+    blockSignPublicKey,
+    first32BytesForgerVrfPublicKey,
+    lastByteForgerVrfPublicKey
+  } = prepareForgerKeys(process.env.WITHDRAW_BLOCK_SIGN_PUBLIC_KEY, process.env.WITHDRAW_VRF_PUBLIC_KEY);
+  const data = contract.methods.withdraw(blockSignPublicKey, first32BytesForgerVrfPublicKey, lastByteForgerVrfPublicKey, value).encodeABI();
+  await signAndSend(web3, contract, address, privateKey, data, 0, gasLimit);
+}
+
+const stakeTotal = async () => {
+  console.log("\nRunning stakeTotal script...");
+  validateEnvVars(
+    "stakeTotal",
+    ["NETWORK"]
+  );
+
+  const stakeTotalBlockSignPublicKey = process.env.STAKE_TOTAL_BLOCK_SIGN_PUBLIC_KEY ? process.env.STAKE_TOTAL_BLOCK_SIGN_PUBLIC_KEY : null;
```

Comment:
I have a doubt regarding the handling of optional parameters. We didn't find a way to define in Solidity optional parameters, so we implemented a convention such that if the parameter has a specific value, it means that it was not actually passed. For bytes parameters (as forger keys or addresses) the specific value is a byte array of 0 (e.g. for address is "0x0000000000000000000000000000000000000000" 20bytes/40 chars, for sign keys is the same but it 32 bytes/64 chars, for vrf keys is 33/bytes/66 chars). For int parameters is 0. 
I have seen that here is used null value instead. Is that correct? I don't know how it is handled by web3 library.

Reply:
I have been following the comment here https://github.com/HorizenOfficial/Sidechains-SDK/blob/7001551b0e6efdc0641db079eb160f19f70edb45/qa/SidechainTestFramework/account/smart_contract_resources/contracts/ForgerStakesV2.sol#L80C5-L91C1 , where it says these values can be null. If there is a specific value to be send as "null", "0x0000000000000000000000000000000000000000" for instance, it would be very helpful to have it there as well.

Reply:
You're right, it will be corrected.